ldap/servers/slapd/opshared.c

```diff
@@ -921,9 +921,7 @@ op_shared_search(Slapi_PBlock *pb, int send_result)
                     next_be = NULL; /* to break the loop */
                     if (operation->o_status & SLAPI_OP_STATUS_ABANDONED) {
                         /* It turned out this search was abandoned. */
-                        pthread_mutex_lock(pagedresults_mutex);
-                        pagedresults_free_one_msgid_nolock(pb_conn, operation->o_msgid);
-                        pthread_mutex_unlock(pagedresults_mutex);
+                        pagedresults_free_one_msgid(pb_conn, operation->o_msgid, pagedresults_mutex);
```

Comment:
IIUC it looks to me that pagedresults_mutex is set if it is an paged result search.
In case of abandon (not paged result) how is it set ?

Reply:
Abandon code does not use `pagedresults_mutex` but  `pageresult_lock_get_addr(pb_conn)`
which will return one of the mutex of the hash array initialized during the server startup
(whether there are paged search result associated with the connection or not ... )

Reply:
Sorry, my question was confusing. The call (pagedresults_free_one_msgid) uses pagedresults_mutex and I thought that pagedresults_mutex was not initialized. But I was wrong, we are in a 'op_is_pagedresults' block where pagedresults_mutex is initialized.
Sorry for the noise